### PR TITLE
Upgrade `itwinjs-core` dependencies to `4.0-dev`

### PR DESCRIPTION
**Pull request by grigasp**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,5 +35,8 @@
   "dependencies": {
     "cross-env": "^7.0.3",
     "npm-run-all": "^4.1.5"
+  },
+  "pnpm": {
```
Comment (saskliutas) [inline package.json:39]: I think this can be removed.

File: app/common/package.json [hunk 2]
```diff
@@ -11,15 +11,17 @@
...
+    "@itwin/eslint-plugin": "^4.0.0-dev.38",
+    "@itwin/core-bentley": "^4.0.0-dev.101",
+    "@itwin/core-common": "^4.0.0-dev.101",
+    "@itwin/core-geometry": "^4.0.0-dev.101",
+    "@itwin/core-quantity": "^4.0.0-dev.101",
+    "@itwin/ecschema-metadata": "^4.0.0-dev.101",
+    "@itwin/ecschema-rpcinterface-common": "^4.0.0-dev.101",
+    "@itwin/presentation-common": "^4.0.0-dev.101"
+  },
+  "devDependencies": {
```
Comment (saskliutas) [inline app/common/package.json:23]: Is there a reason why `eslint` and `typescript` were moved under devDependencies?

**Review by saskliutas — APPROVED** [review]

File: app/common/package.json (hunk 2 above)
Comment (grigasp) [reply app/common/package.json]: probably not, merged them back to `dependencies`

File: package.json (hunk 1 above)
Comment (grigasp) [reply package.json]: removed

**Pull request merged**